src/hdmf/container.py

```diff
@@ -273,6 +274,20 @@ def generate_new_id(self, **kwargs):
             for c in self.children:
                 c.generate_new_id(**kwargs)
 
+    @docval({'name': 'term', 'type': str, 'doc': "term to be validated"},
+            {'name': 'term_set', 'type': TermSet, 'doc': 'the set of terms used to validate data on add'})
+    def validate_data(self, **kwargs):
+            """
```

Comment:
If a user must provide both the `term` and `term_set`, what is the utility of having this as part of the `AbstractContainer` class? Wouldn't it be simpler to have a `validate_term` method on the `TermSet`, i.e., call `term_set.validate(term)` instead.

Reply:
I think it would be the same level of complexity. Both are just term lookups. Having it on AbstractContainer could help us down the line expanding it to attributes, but I don't think a change is warranted.

Reply:
Being on AbstractContainer allows, validate to be used outside of just Data, possibly with attributes. However, thus far Data is the only one using it and even though it stores the termset, I am not sure to remove the termset parameter. Are you suggesting moving it to the TermSet class and everything (both now and future) will call validate from there? I think I see the simplification that comes from that. I'll start implementing and see if there's any issues we didn't see.